contracts/utils/BridgeMessage.sol

```diff
@@ -2,33 +2,12 @@
 pragma solidity ^0.8.20;
 
 /// @title BridgeMessage
-/// @notice This library defines the message format and constants for the Sui native bridge.
-/// @dev The message prefix and the token decimals are fixed for the Sui bridge.
+/// @notice This library defines the message format and constants for the Sui native bridge. It also
+/// provides functions to encode and decode bridge messages and their payloads.
+/// @dev This library only utilizes internal functions to enable upgradeability via the OpenZeppelin
+/// UUPS proxy pattern (external libraries are not supported).
```

Comment:
libraries with "external" decorated functions are not supported by the OZ upgradeable library atm. only "internal" decorated functions are supported because when compiled, they are just appended to the importing contract.

Reply:
interesting, why is library special?

Reply:
libraries compile differently depending on internal/external